### PR TITLE
Add username/password in Metricbeat autodiscover hints

### PR DESCRIPTION
**Pull request by ChrsMark**
This PR closes https://github.com/elastic/beats/issues/15115

### Manual Testing
1. Enable autodiscover in metricbeat.yml
```
metricbeat:
  autodiscover.providers:
    - type: docker
      hints.enabled: true
```
2. Start metricbeat with log enabled: `./metricbeat -e -d "hints.builder" `
3. `docker run -l co.elastic.metrics/module=nats -l co.elastic.metrics/username=user42 --name nats nats`
4. Check in Metricbeat logs for : `2019-12-16T15:53:18.712+0200	DEBUG	[hints.builder]	hints/metrics.go:144	generated config: {"enabled":true,"hosts":null,"metricsets":["connections","routes","stats","subscriptions"],"module":"nats","period":"1m","processors":null,"ssl":null,"timeout":"3s","username":"user42"}`

### CODE REVIEW

File: metricbeat/autodiscover/builder/hints/metrics.go [hunk 1]
```diff
@@ -198,6 +208,14 @@ func (m *metricHints) getMetricPath(hints common.MapStr) string {
 	return builder.GetHintString(hints, m.Key, metricspath)
 }
 
+func (m *metricHints) getUserName(hints common.MapStr) string {
```
Comment (mtojek) [inline metricbeat/autodiscover/builder/hints/metrics.go]: nit pick only: isn't it correct in English to use a single word "username" instead two ones "user Name"?

File: metricbeat/docs/autodiscover-hints.asciidoc [hunk 2]
```diff
@@ -35,6 +35,16 @@ The time interval for metrics retrieval, ie: 10s
...
 
+[float]
+===== `co.elastic.metrics/username`
+
+The username to use for authentication
+
+[float]
+===== `co.elastic.metrics/password`
+
+The password to use for authentication
```
Comment (exekias) [inline metricbeat/docs/autodiscover-hints.asciidoc]: I think we should put a note here advising not to use plain text passwords here, but references to ENV vars in the Metricbeat container, or passwords stored in keystore. Long term I think we should implement a way to allow referencing passwords from k8s secrets

**Comment by ycombinator** [comment]
Tested this functionality manually with BC1 and it works as expected. However, I do have one concern. When the password is provided via a container label it is emitted in the Metricbeat debug log. Should we omit it or mask it? Or is it okay since it's not emitted in the Metricbeat log by default but only when debug level logging is turned on?

**Comment by ycombinator** [comment]
@ChrsMark pointed me to a related discussion that already happened in this PR: https://github.com/elastic/beats/pull/15349#discussion_r364682852. So this is a known issue and user should specify the password carefully.

**Comment by ycombinator** [comment]
I put up a PR to mask the password in log output: https://github.com/elastic/beats/pull/15616.

**Pull request merged**